packages/babel-parser/src/parser/expression.js

```diff
@@ -1138,6 +1138,19 @@ export default class ExpressionParser extends LValParser {
           this.registerTopicReference();
           return this.finishNode(node, "PipelinePrimaryTopicReference");
         }
+
+        if (this.hasPlugin("privateIn")) {
+          node = this.parseMaybePrivateName(true);
+          this.classScope.usePrivateName(node.id.name, node.start);
```

Comment:
Can you add a new test case for `#constructor in c`? We check `#constructor` only in `parseClassPropertyName`.